### PR TITLE
Fixes #3384: deprecate jQuery.camelCase

### LINKED ISSUE #3384: Deprecate jQuery.camelCase, hide in 4.0
### Description ###

We should hide `jQuery.camelCase`. It's an undocumented, private method which doesn't make sense to be available externally as it contains a CSS-only quirk that replaces `-ms-` to `ms-`. Rather than fixing it it's better to just remove it.

Theoretically we could do it in any minor release but since it's been semi-public for so long it's better to wait, especially that code around it will have to change in 4.0 anyway because of #3355.

### Link to test case ###
N/A

### PR DESCRIPTION
**Pull request by timmywil**
### Summary ###
Makes adjustments to #3604 to fix #3384.

### Checklist ###

* [x] All authors have signed the CLA at https://cla.js.foundation/jquery/jquery
* [x] New tests have been added to show the fix or feature works
* [x] Grunt build and unit tests pass locally with these changes
* [x] If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com

### CODE REVIEW

File: src/manipulation.js [hunk 1]
```diff
@@ -45,7 +45,6 @@ var
 
 	// checked="checked" or checked
 	rchecked = /checked\s*(?:[^=]|=\s*.checked.)/i,
-	rscriptTypeMasked = /^true\/(.*)/,
```
Comment (dmethvin) [inline src/manipulation.js]: The changes in this file seem unrelated to the PR?

Comment (timmywil) [reply src/manipulation.js]: Indeed. I didn't see that before. Nice catch!

**Comment by timmywil** [comment]
Merged in https://github.com/jquery/jquery/commit/775caebd617e91e697b3e41ae33d6877e63785aa
